core/includes/bootstrap.inc

```diff
@@ -623,9 +623,10 @@ function find_conf_path($http_host, $script_name, $require_settings = TRUE) {
     for ($i = count($uri) - 1; $i > 0; $i--) {
       for ($j = count($server); $j > 0; $j--) {
         $dir = implode('.', array_slice($server, -$j)) . implode('.', array_slice($uri, 0, $i));
-        if (isset($sites[$dir]) && file_exists(BACKDROP_ROOT . '/sites/' . $sites[$dir])) {
-          $dir = $sites[$dir];
-        }
+        if (isset($sites[$dir])) {
+          $conf = './sites/' . $sites[$dir];
+          return $conf;
+       }
         if (file_exists(BACKDROP_ROOT . '/sites/' . $dir . '/settings.php') || (!$require_settings && file_exists(BACKDROP_ROOT . '/sites/' . $dir))) {
           $conf = './sites/' . $dir;
           return $conf;
```

Comment:
Why is not the `file_exists()` call removed from these lines too?

Reply:
Not sure. This PR is a direct backport of the latest patch available in https://www.drupal.org/project/drupal/issues/1055856, and this the only bit that is changed there as well.

I am not claiming to understand everything around this - just want to check if there is in fact a performance improvement as suggested over in that issue.